packages/template-retail-react-app/app/components/_app/index.jsx

```diff
@@ -327,4 +343,50 @@ App.propTypes = {
     config: PropTypes.object
 }
 
+/**
+ * a hook that parallelly and individually fetches category based on the given ids
+ * @param ids - list of categories ids to fetch
+ * @param queryOptions -  react query options
+ * @return list of react query results
+ */
+export const useCategoryBulk = (ids, queryOptions) => {
+    if (!ids || ids.length === 0) {
+        return
```

Comment:
What happens if we do `useQueries({queries: []})`? We might want to consistently return the hook interface, rather than sometimes returning `undefined`.

Reply:
We  can set enable for the queries with !!id 🤔